### DISCUSSION TITLE
Select cached data without using hooks

### DISCUSSION THREAD

**Discussion by tommytritran** [reduxjs/redux-toolkit#2488] [Q&A] [answered]
I'm trying to select cached data from RTKQ without using the auto-generated query hook, but I'm having trouble understanding the [docs](https://redux-toolkit.js.org/rtk-query/usage/usage-without-react-hooks#:~:text=With%20React%20hooks%2C%20this%20behaviour%20is%20instead%20handled).

```
const result = api.endpoints.getPosts.select()(state)
const { data, status, error } = result
```
This is how the docs describe how to access the data, but I can't find any references on how to inject the state object "select()(state)".

I can't figure out how to access the data if I only call the select?

`api.endpoints.getPosts.select()`

Can someone explain me the difference between "select()" and "select()(state)"

Or what is the optimal solution to access the cached data from RTKQ?

**Comment by phryneas** [comment, marked as answer]
Answered [on StackOverflow](https://stackoverflow.com/questions/72869959/rtkq-select-data-without-hooks)
